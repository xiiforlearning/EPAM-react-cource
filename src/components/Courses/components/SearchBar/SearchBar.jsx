import React from 'react';
import { Button } from '../../../../common/Button/Button';
import Input from '../../../../common/Input/Input';
import {
	BUTTON_TEXT_ADD_COURSE,
	BUTTON_TEXT_SEARCH,
} from '../../../../constants';
import styles from './searchBar.module.scss';

const SearchBar = ({ addCourseOnClick }) => {
	return (
		<div className={styles.wrapper}>
			<form
				className={styles.form}
				onSubmit={(event) => {
					event.preventDefault();
				}}
			>
				<Input />
				<Button text={BUTTON_TEXT_SEARCH}></Button>
			</form>
			<Button text={BUTTON_TEXT_ADD_COURSE} onClick={addCourseOnClick}></Button>
		</div>
	);
};

export default SearchBar;
