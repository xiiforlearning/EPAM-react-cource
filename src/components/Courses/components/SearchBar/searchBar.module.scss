.wrapper {
    display: flex;
    justify-content: space-between;
    width: 100%;
}
.form {
    width: 50%;
    display: flex;
    justify-content: space-between;
    gap: 20px;

    input {
        flex: 4;
    }

    button {
        flex: 1;
    }
} 