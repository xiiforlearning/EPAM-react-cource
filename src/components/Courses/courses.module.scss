.wrapper {
    padding: 20px;
    border: 1px solid blue;
    display: flex;
    flex-direction: column;
    gap: 30px;
}