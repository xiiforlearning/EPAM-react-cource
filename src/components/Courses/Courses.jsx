import { useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';

import CourseCard from './components/CourseCard/CourseCard';
import SearchBar from './components/SearchBar/SearchBar';

import styles from './courses.module.scss';

const Courses = () => {
	const navigate = useNavigate();
	const courses = useSelector((state) => state.course);
	const allAuthors = useSelector((state) => state.author);

	return (
		<div className={styles.wrapper}>
			<SearchBar
				addCourseOnClick={() => {
					navigate('/courses/add');
				}}
			/>
			{courses.map((item) => (
				<CourseCard
					key={item.id}
					title={item.title}
					description={item.description}
					creationDate={item.creationDate}
					duration={item.duration}
					authors={item.authors}
					allAuthors={allAuthors}
					id={item.id}
				/>
			))}
		</div>
	);
};

export default Courses;
