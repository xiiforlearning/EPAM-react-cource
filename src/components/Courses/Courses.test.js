import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import { mockedAuthorsList, mockedCoursesList } from '../../constants';
import SearchBar from './components/SearchBar/SearchBar';
import Courses from './Courses';

const mockedState = {
	user: {
		role: 'admin',
		isAuth: true,
	},
	course: mockedCoursesList,
	author: mockedAuthorsList,
};

const mockedStore = {
	getState: () => mockedState,
	subscribe: jest.fn(),
	dispatch: jest.fn(),
};

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useNavigate: () => mockedUsedNavigate,
}));

describe('Courses test', () => {
	it('should render Courses component properly', () => {
		const component = create(
			<Provider store={mockedStore}>
				<Courses />
			</Provider>
		);

		expect(component.toJSON()).toMatchSnapshot();
	});

	it('courseform should be showed after a click on a button "Add New Course"', async () => {
		const component = create(
			<Provider store={mockedStore}>
				<SearchBar />
			</Provider>
		);

		expect(component.toJSON()).toMatchSnapshot();
	});
});
