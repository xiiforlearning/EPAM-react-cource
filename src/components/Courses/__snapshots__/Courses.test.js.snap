// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Courses test courseform should be showed after a click on a button "Add New Course" 1`] = `
<div
  className="wrapper"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <label>
      <input
        placeholder="Enter undefined..."
        type="text"
      />
    </label>
    <button>
      Search
    </button>
  </form>
  <button>
    Add New Course
  </button>
</div>
`;

exports[`Courses test should render Courses component properly 1`] = `
<div
  className="wrapper"
>
  <div
    className="wrapper"
  >
    <form
      className="form"
      onSubmit={[Function]}
    >
      <label>
        <input
          placeholder="Enter undefined..."
          type="text"
        />
      </label>
      <button>
        Search
      </button>
    </form>
    <button
      onClick={[Function]}
    >
      Add New Course
    </button>
  </div>
  <div
    className="wrapper"
    data-testid="course"
  >
    <div
      className="courseDesc"
    >
      <h2>
        JavaScript
      </h2>
      <p>
        Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum 
                    has been the industry's standard dummy text ever since the 1500s, when an unknown 
                    printer took a galley of type and scrambled it to make a type specimen book. It has survived 
                    not only five centuries, but also the leap into electronic typesetting, remaining essentially u
                    nchanged.
      </p>
    </div>
    <div
      className="courseInfoWrapper"
    >
      <div
        className="courseInfo"
      >
        <div>
          <span>
            Autors: 
          </span>
          <p>
            Vasiliy Dobkin, Nicolas Kim
          </p>
        </div>
        <div>
          <span>
            Duration: 
          </span>
          <p>
            2 : 40 hours
          </p>
        </div>
        <div>
          <span>
            Created: 
          </span>
          <p>
            08.03.2021
          </p>
        </div>
      </div>
      <div
        className="buttonsWrapper"
      >
        <button
          onClick={[Function]}
        >
          Show Course
        </button>
        <button
          onClick={[Function]}
        >
          delete
        </button>
        <button
          onClick={[Function]}
        >
          update
        </button>
      </div>
    </div>
  </div>
  <div
    className="wrapper"
    data-testid="course"
  >
    <div
      className="courseDesc"
    >
      <h2>
        Angular
      </h2>
      <p>
        Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum 
                    has been the industry's standard dummy text ever since the 1500s, when an unknown 
                    printer took a galley of type and scrambled it to make a type specimen book.
      </p>
    </div>
    <div
      className="courseInfoWrapper"
    >
      <div
        className="courseInfo"
      >
        <div>
          <span>
            Autors: 
          </span>
          <p>
            Anna Sidorenko, Valentina Larina
          </p>
        </div>
        <div>
          <span>
            Duration: 
          </span>
          <p>
            3 : 30 hours
          </p>
        </div>
        <div>
          <span>
            Created: 
          </span>
          <p>
            10.11.2020
          </p>
        </div>
      </div>
      <div
        className="buttonsWrapper"
      >
        <button
          onClick={[Function]}
        >
          Show Course
        </button>
        <button
          onClick={[Function]}
        >
          delete
        </button>
        <button
          onClick={[Function]}
        >
          update
        </button>
      </div>
    </div>
  </div>
</div>
`;
